cores/esp32/HardwareSerial.h

```diff
@@ -138,6 +150,7 @@ class HardwareSerial: public Stream
     uart_t* _uart;
     size_t _rxBufferSize;
     OnReceiveCb _onReceiveCB;
+    uint8_t _onReceiveTimeout;
```

Comment:
this may not be necessary to exist anymore as per the suggestions already made here.

Reply:
I will stress my previous comments here. Because you need to call uart_set_rx_timeout after the uart initialization, then I need a local variable to store my timeout config in case setRxTimeout was called before begin().